### ISSUE TITLE
Differences docker-toolbox / kmachine

### ISSUE THREAD

**Issue by tdeheurles** [skippbox/kmachine#32]
I was asking myself on the differences between your `kmachine` and the `docker-toolbox` project.

What are the differences with :
```
install docker-toolbox
start a kubernetes node using the docker version
connect to it
```
and your installation ?
Is it all the automated deployment on all the provider ?

Sorry, don't feel attacked by this question, I just want to understand ...

**Comment by runseb** [comment]
`kmachine` is a fork of `docker-machine`.
Instead of using compose to provision k8s onto a docker-machine, or some bash script, we put the provisioning inside the docker-machine code.

That way you have one binary, totally similar to docker-machine, but it gives you k8s.

We also have kompose, a UI and boot2k8s. So technically we have all the components of a k8s toolbox which mimics docker-toolbox, but adds k8s support.

Does that make sense ?

**Comment by tdeheurles** [comment]
Yes, It totally makes sense.
We are all trying to solve this kind of things in different ways :D

I'm working on [this project](https://github.com/tdeheurles/hive) and I'm looking how to integrate a local kubernetes in it.
Our project is not similar but goes in the same way. You deploy a kubernetes (in need the virtualbox part).
My project try to manage all the stuff around developing and deploying application from a developer point of view. I set some pattern on how to put some files, and then I can build/push/deploy the application with some CLI execution.
Where you have wrapped lots of things in the executable, I try to manage everything from containers (kubectl, gcloud cli running from containers, build/push commands run also from container).

We are really close but don't take the same path... I'm today asking myself if I ask the developer to install kmachine or if I add a `./hive start kubernetes` command (using docker-toolbox).

**Comment by tdeheurles** [comment]
Understand that I had bad time integrating docker on windows as the terminal have lots of issues. Your project doesn't come with this `Docker Quick Terminal` that solves most of the windows issues (And it won't make sense to integrate it in kmachine).

**Comment by runseb** [comment]
kmachine is only a part of what will be the k8s toolbox. We haven't had tim unfortunately to work on the windows portion and create the Quick terminal .

Maybe we can collaborate, and make kmachine do what you need on windows ?

**Comment by tdeheurles** [comment]
What is k8s toolbox ? Which are the teams behind ?

**Comment by tdeheurles** [comment]
Yes and no ... I really like the idea of one virtual machine bringing docker and k8s. I think it's the good solution. But on the other hand I don't really know what are the requirement to make it works on Windows ...

The point is that the docker-toolbox (for windows) team have passed a lot of time to have a stable release and I'm not sure that duplicating that work is a good idea ...
`Windows is a ... in the ... ` oO ... :-/

**Comment by runseb** [comment]
@tdeheurles sorry for the delayed response.
k8s toolbox is one of the things we work on at Skippbox (new container startup) it is made of:

* kmachine
* kompose
* boot2k8s
* skippbox (a k8s desktop UI).

They are inspired by the docker toolbox and some tools are forks or variant.

We have not finished the packaging and not tested any of the windows setup which explains the issues you are seeing.

**Comment by tdeheurles** [comment]
Ok, thank you.

**Comment by runseb** [comment]
Salut @tdeheurles désolé pour le long retard. Je te presente @gregbkr qui parle Francais et connait Windows bien mieux que moi. Il commence a regarder kmachine sur windows.
